File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_BillboardFrame.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
#/==========================================================================
#*!
#	@brief	t_BillboardFrame
#/
class t_BillboardFrame:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_texid = 0	#U32
	m_color = 0	#U32
	m_frame = 0.0	#float
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_texid = 0
		self.m_color = 0
		self.m_frame = 0.0

	def read(self,cVariable):
		self.m_texid = cVariable.getU32()
		self.m_color = cVariable.getU32()
		self.m_frame = cVariable.getFloat()
		return True

	def write(self,cVariable):
		cVariable.putU32(self.m_texid)
		cVariable.putU32(self.m_color)
		cVariable.putFloat(self.m_frame)
		return True

File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_BillboardFrame.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_BillboardFrame
*/
public class t_BillboardFrame : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public uint	m_texid;
	public uint	m_color;
	public float	m_frame;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_BillboardFrame() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_texid = 0;
		m_color = 0;
		m_frame = 0.0f;
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_texid = cVariable.getU32();
			m_color = cVariable.getU32();
			m_frame = cVariable.getFloat();
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_texid);
		cVariable.put(m_color);
		cVariable.put(m_frame);
		return true;
	}
};

